draft-ietf-teep-protocol.md

```diff
@@ -1392,6 +1389,18 @@ The cipher suites defined above do not do encryption at the TEEP layer, but
 permit encryption of the SUIT payload (e.g., using {{I-D.ietf-suit-firmware-encryption}}).
 See {{security}} for more discussion of specific payloads.
 
+For the initial QueryRequest message, unless the TAM has more specific knowledge about the TEEP Agent
+(e.g., if the QueryRequest is sent in response to some underlying transport message that contains a hint),
+the message does not use one of the above cipher suites but instead uses COSE_Sign with multiple signatures,
+one for each algorithm used in any of the cipher suites listed in the supported-cipher-suites
+parameter of the QueryRequest, so that a TEEP Agent supporting any one of them can verify a signature.
+
+For an Error message with code ERR_UNSUPPORTED_CIPHER_SUITES, the TEEP Agent MUST
+protect it with one of the cipher suites mandatory for the TAM.
```

Comment:
Because an un-supported cipher error can be reported, the discussion above is revised as "SHOULD" for multiple signatures, not mandating it.

Reply:
I don't follow.  This is saying the TEEP Agent MUST use one of the two mandatory cipher suites, which earlier text already says the agent can pick one of to implement.  It knows that any compliant TAM will be able to check the signature on the Error as a result.   This case can only happen if the TAM doesn't list either of the mandatory cipher suites in the supported-cipher-suites list in the QueryRequest.  But the QueryRequest message defines that field as
"The supported-cipher-suites parameter lists the cipher suites supported by the TAM."
and this section requires the mandatory ones to be supported, so this is just there for future compatibility if we later have a newer spec that a TAM implements that changes the list.

Reply:
I meant that the multiple signature from TAM request message should be "SHOULD", not necessary to make it a "MUST" when we have this error to fallback to.

Reply:
Is there any technical argument against the MUST?   The MUST helps efficiency and interoperability I think.

As an aside: If a TEEP Agent gets a message with no signature it can check, then it can't verify whether it was actually sent by a trusted TAM, and so needs to be careful not to put any sensitive information in the response.   The Error here would contain the Agent's "supported-cipher-suites" which is currently mandatory to include if err-code is ERR_UNSUPPORTED_CIPHER_SUITES.  Hopefully we don't believe that the Agent's "supported-cipher-suites" is sensitive information.